lumeer-core/src/main/java/io/lumeer/core/facade/SearchFacade.java

```diff
@@ -60,17 +60,80 @@
    public List<Collection> searchCollections(Query query) {
       Set<Collection> collections = new HashSet<>();
 
-      if (!isCollectionSelectionOnly(query)) {
-         collections.addAll(getCollectionsByDocumentSearch(query));
+      if (query.getFulltext() != null && !query.getFulltext().isEmpty()) {
+         collections.addAll(getCollectionsByDocumentsSearch(query));
       }
 
-      collections.addAll(getCollectionsByCollectionSearch(query));
+      if (query.getDocumentIds() != null && !query.getDocumentIds().isEmpty()) {
+         collections.addAll(getCollectionsByDocumentsIds(query.getDocumentIds()));
+      }
+
+      if (isOnlyCollectionQuery(query) || collectionQueryIsNotEmpty(query)) {
+         collections.addAll(getCollectionsByCollectionSearch(query));
```

Comment:
Why this strange condition? I think there should either be AND operator or otherwise the first part of the condition does not make sense. Because if it is true the second one will also be true so the first part does not matter.

Reply:
The first part of conditions will ensure, that empty query returns all collections.

Reply:
What? Do we want to run `getCollectionsByCollectionSearch(query)` if the query is empty (first condition) and also if it is not empty (second condition)? Why do we then have this `if` statement?

Reply:
Ok, maybe names of conditions are confusing, so I will change it.

Reply:
Is it better now?

Reply:
Oh, yes. I totally understand it now! Thanks.